jaeger-spark-dependencies-elasticsearch/src/main/java/io/jaegertracing/spark/dependencies/elastic/ElasticsearchDependenciesJob.java

```diff
@@ -0,0 +1,242 @@
+    }
+    conf.set("es.nodes", parseHosts(builder.hosts));
+    if (builder.hosts.indexOf("https") != -1) {
+      conf.set("es.net.ssl", "true");
+    }
+    for (Map.Entry<String, String> entry : builder.sparkProperties.entrySet()) {
+      conf.set(entry.getKey(), entry.getValue());
+    }
+  }
+
+  public void run() {
+    run(index + "-span-" + dateStamp,"jaeger-dependencies-" + dateStamp + "/dependencies");
+    log.info("Done");
+  }
+
+  void run(String spanResource, String depResource) {
+    JavaSparkContext sc = new JavaSparkContext(conf);
+    try {
+      JavaPairRDD<String, Iterable<Span>> traces = JavaEsSpark.esJsonRDD(sc, spanResource)
+          .map(new ElasticTupleToSpan())
+          .groupBy(Span::getTraceId);
+
+      traces.foreach(stringIterableTuple2 -> {
+        System.out.println(stringIterableTuple2._1() +"  ->  " +  Lists.newArrayList(stringIterableTuple2._2()).size());
+      });
+
+      List<Dependency> dependencyLinks = DepenencyLinksSparkJob.derive(traces);
+      store(sc, dependencyLinks, depResource);
+    } finally {
+      sc.stop();
+    }
+  }
+
+  private void store(JavaSparkContext javaSparkContext, List<Dependency> dependencyLinks, String resource) {
+    if (dependencyLinks.isEmpty()) {
+      return;
+    }
+
+    String json;
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      json = objectMapper.writeValueAsString(new ElasticsearchDependencies(dependencyLinks, day));
+    } catch (JsonProcessingException e) {
+      throw new IllegalStateException("Could not serialize dependencies", e);
+    }
+
+    JavaEsSpark.saveJsonToEs(javaSparkContext.parallelize(Collections.singletonList(json)), resource);
+  }
+
+  static String parseHosts(String hosts) {
+    StringBuilder to = new StringBuilder();
+    String[] hostParts = hosts.split(",");
+    for (int i = 0; i < hostParts.length; i++) {
+      String host = hostParts[i];
+      if (host.startsWith("http")) {
+        URI httpUri = URI.create(host);
+        int port = httpUri.getPort();
+        if (port == -1) {
+          port = host.startsWith("https") ? 443 : 80;
+        }
+        to.append(httpUri.getHost() + ":" + port);
+      } else {
+        to.append(host);
+      }
+      if (i + 1 < hostParts.length) {
+        to.append(',');
+      }
+    }
+    return to.toString();
+  }
+
+  /**
+   * Helper class used to serialize dependencies to JSON.
+   */
+  public static final class ElasticsearchDependencies {
+    private List<Dependency> dependencies;
+    private long ts;
+
+    public ElasticsearchDependencies(List<Dependency> dependencies, long ts) {
+      this.dependencies = dependencies;
+      this.ts = ts;
+    }
+
+    public List<Dependency> getDependencies() {
+      return dependencies;
+    }
+
+    public String getTimestamp() {
+      // Jaeger ES dependency storage uses RFC3339Nano for timestamp
+        return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX")
```

Comment:
Isn't there a constant for this format already?

Reply:
I couldn't find any.

Reply:
Again, Java 8 could have helped: 
https://docs.oracle.com/javase/8/docs/api/java/time/OffsetDateTime.html

Reply:
(btw, I just noticed now that the method name is misleading: this is not returning a timestamp, but a date as string).